pkg/pilot/elasticsearch/v5/options.go

```diff
@@ -144,8 +144,15 @@ func (o *PilotOptions) Run(stopCh <-chan struct{}) error {
 		return err
 	}
 
-	// start the shared informer factories
-	go o.sharedInformerFactory.Start(stopCh)
+	// set the genericPilot on
+	o.pilot.genericPilot = genericPilot
+
+	// create a new stopCh just for the factory so the factory continues to
+	// receive updates after the process has been signaled to exit
+	stopInformers := make(chan struct{})
+	defer close(stopInformers)
+	// start the shared informer factory
+	go o.sharedInformerFactory.Start(stopInformers)
```

Comment:
❓ So this is the case where kubelet has sent sigterm to the pilot, but we need to continue running while we evict shards from this database node etc and update the Pilot status.
Perhaps expand a little on the comment above.